src/float.satyg

```diff
@@ -95,6 +100,31 @@ end = struct
   let of-int = float
   let to-int = round
 
+  let of-string-opt str =
+    let dot = Char.make `.` in
+    let (first-str, second-str-opt) = String.split-by-first dot str in
+    let second-str =
+      match second-str-opt with
+      | Some(s) -> s
+      | None -> `0`
+    in
+    let first-int-opt = Int.of-string-opt first-str in
+    let second-int-opt = Int.of-string-opt second-str in
+    let second-str-len = String.length second-str in
+    let-rec make-fl n fl =
+      if Int.(n <= 0) then
+        fl
+      else
+        make-fl (n - 1) (fl *. 0.1)
```

Comment:
I'd recommend to avoid multiplying 0.1, which can not be exactly represented in IEEE 754 binary formats, so many times. See https://github.com/ulfjack/ryu/blob/master/ryu/s2d.c#L67 for reference.

Reply:
Please let me know the appropriate correction.
I have a lack of knowledge.

Reply:
Although implementing that in the correct way is hard, I believe that parsing both integer and decimal parts in integer and dividing the decimal part by `10^{length of decimal}` should produce better results than current one.

We could work on more precise implementation later..

Reply:
Thanks for updating!!